src/main/java/com/jackshenorion/smarts/util/io/IOUtil.java

```diff
@@ -525,7 +525,9 @@ public static void writeAndClose(File file, byte[] data) {
 
```

Comment:
**Reporter**: FINDBUGS
**Rule**: RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE
**Severity**: WARN
**File**: src/main/java/com/jackshenorion/smarts/util/io/IOUtil.java L462
**Source**: com.jackshenorion.smarts.util.io.IOUtil

Nullcheck of value previously dereferenced


<p> A value is checked here to see whether it is null, but this value can't
be null because it was previously dereferenced and if it were null a null pointer
exception would have occurred at the earlier dereference.
Essentially, this code and the previous dereference
disagree as to whether this value is allowed to be null. Either the check is redundant
or the previous dereference is erroneous.</p>


*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<564642453>*